tests/Client/LogTest.php

```diff
@@ -0,0 +1,219 @@
+<?php
+
+namespace HelloFresh\Stats\Client;
+
+
+use HelloFresh\Stats;
+use HelloFresh\Stats\Bucket;
+use HelloFresh\Stats\Bucket\MetricOperation;
+use HelloFresh\Stats\Bucket\Plain;
+use HelloFresh\Stats\Incrementer;
+use HelloFresh\Stats\State;
+use PHPUnit\Framework\TestCase;
+
+class LogTest extends TestCase
+{
+    public function setUp()
+    {
+        mt_srand(time());
+    }
+
+    public function testInstances()
+    {
+        /** @var \PHPUnit_Framework_MockObject_MockObject|\Psr\Log\LoggerInterface $logger */
+        $logger = $this->getMockBuilder('\Psr\Log\LoggerInterface')->getMock();
+
+        $statsClient = new Log($logger);
+        $this->assertInstanceOf(Stats\Timer\Log::class, $statsClient->buildTimer());
+
+        $reflection = new \ReflectionClass($statsClient);
+
+        $methodIncrementer = $reflection->getMethod('getIncrementer');
+        $methodIncrementer->setAccessible(true);
+        $this->assertInstanceOf(Incrementer\Log::class, $methodIncrementer->invoke($statsClient));
```

Comment:
:-1: You should not test none public methods. In this case you could create a manual mock that makes the methods public but I would advice against this since this method should be tested based on other units taht call it internally. (same go's for all other reflection calls)